### PR TITLE
Actions caching

### PR DESCRIPTION
**Pull request by kyscott18**
Utilize caching in order to speed up some actions

### CODE REVIEW

**Comment by kyscott18** [comment]
After looking at some actions in practice, the extra caching seems like it sped up the setup step by about 15 seconds (25%). Not great

File: examples/with-docker/package.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
     "lint": "eslint ."
   },
   "dependencies": {
-    "@ponder/core": "latest"
```
Comment (kyscott18) [inline examples/with-docker/package.json:10]: @0xOlias Is this a mistake?

Comment (0xOlias) [reply examples/with-docker/package.json:10]: I think so. NGL, I haven't ran this docker example in a while so not certain it's working at all.

**Review by 0xOlias — APPROVED** [review]
LGTM

**Pull request merged**